features/device-config/monitor/src/main/java/org/opennms/features/deviceconfig/monitors/DeviceConfigMonitor.java

```diff
@@ -87,15 +93,31 @@
         final Map<String, Object> params = new HashMap<>();
         final String configType = getKeyedString(parameters, CONFIG_TYPE, ConfigType.Default);
         final OnmsIpInterface ipInterface = ipInterfaceDao.findByNodeIdAndIpAddress(svc.getNodeId(), svc.getIpAddr());
-        final Optional<org.opennms.features.deviceconfig.persistence.api.DeviceConfig> deviceConfigOptional = deviceConfigDao.getLatestConfigForInterface(ipInterface, configType);
+        final var deviceConfigOptional = deviceConfigDao.getLatestConfigForInterface(ipInterface, configType);
+        deviceConfigOptional.ifPresent(deviceConfig -> params.put(LAST_RETRIEVAL, String.valueOf(deviceConfig.getLastUpdated().getTime())));
 
-        if (deviceConfigOptional.isPresent()) {
-            params.put(LAST_RETRIEVAL, String.valueOf(deviceConfigOptional.get().getLastUpdated().getTime()));
+        final String scriptFile = getKeyedString(parameters, SCRIPT_FILE, null);
+        try {
+            if (scriptFile != null) {
+                String script = parseScriptFile(scriptFile);
+                params.put(SCRIPT, script);
+            }
+        } catch (Exception e) {
+            LOG.error("Error while parsing script file {}", scriptFile, e);
```

Comment:
Should we re-throw here? This would give a better error message than a NPE in the poller.